utils.c

```diff
@@ -593,6 +593,34 @@ get_file_mode (const char *pathname)
   return buf.st_mode & S_IFMT;
 }
 
+int
+ensure_dir (const char *path,
+            mode_t      mode)
+{
+  struct stat buf;
+
+  /* We check this ahead of time, otherwise
+     the mkdir call can fail in the read-only
+     case with EROFS instead of EEXIST on some
+     filesystems (such as NFS) */
+  if (stat (path, &buf) == 0)
```

Comment:
could the `stat` check be done only in the case `mkdir` fails with EROFS?

Reply:
Sure, but there might be other similar errors, so it seems better to do the stat. I think the common case is that the targe exists, so performance is similar (ie one syscall)